backend/src/apiserver/server/pipeline_upload_server.go

```diff
@@ -235,12 +260,15 @@ func (s *PipelineUploadServer) UploadPipelineVersion(w http.ResponseWriter, r *h
 
 	w.Header().Set("Content-Type", "application/json")
 	marshaler := &jsonpb.Marshaler{EnumsAsInts: false, OrigName: true}
-	createdPipelineVersion := toApiPipelineVersionV1(newPipelineVersion)
-	if createdPipelineVersion == nil {
-		s.writeErrorToResponse(w, http.StatusInternalServerError, util.NewInternalServerError(errors.New("Failed to convert internal pipeline version representation to its v1beta1 API counterpart"), "Failed to create a pipeline version due to error converting it to API"))
+	if api_version == "v1beta1" {
+		err = marshaler.Marshal(w, toApiPipelineVersionV1(newPipelineVersion))
```

Comment:
In the previous code, we have a check for the output of `toApiPipelineVersionV1()` to be `nil' or not. I wonder if we should keep it.

Reply:
`toApiPipelineVersionV1()` per its [current implementation](https://github.com/kubeflow/pipelines/blob/01c87f8a032e70a6ca92cdbefa974a7da387f204/backend/src/apiserver/server/api_converter.go#L432) never returns nil. So the previous code wasn't useful.